### ISSUE TITLE
`dom-node-dataset`: false-negative with `dataset` reassignment

### ISSUE THREAD

**Issue by silverwind** [sindresorhus/eslint-plugin-unicorn#3262]
Using `{preferAttributes: true}`, this edge case is missed:

```js
// ❌
const data = el.dataset; // should raise error here
console.log(data.fooBar);
console.log(data.bazQux);

// ✅
console.log(el.getAttribute('data-foo-bar'));
console.log(el.getAttribute('data-baz-qux'));
```
